app/routes/ConfirmTransaction/containers/ConfirmTransaction.js

```diff
@@ -20,95 +24,151 @@ class ConfirmTransaction extends Component {
     this.password = validPassword ? location.state.password : undefined
 
     this.state = {
-      transaction: {},
+      transactionNumber: 0,
+      balance: undefined,
+      unlockRequest: false,
+      reviewedTx: false,
     }
   }
 
   componentDidMount = () => {
-    chrome.windows.getCurrent((window) => {
-      this.showTransaction(window.id)
-    })
+    const { transactionNumber } = this.state
+    this.showTransaction(transactionNumber)
   }
 
-  showTransaction = (windowId) => {
-    const { transactions } = this.props
-    if (!transactions || transactions.length === 0) {
-      return
-    }
+  getBalance = async (address) => {
+    const web3 = new Web3(new Web3.providers.HttpProvider(config.networks[config.currentNetwork].url))
```

Comment:
TODO, for next version, I suggest to extract the creation of web3 objects to a singleton instance.